### PR TITLE
Gogs chart

### PR DESCRIPTION
**Pull request by tvon**
This adds a chart for Gogs (https://github.com/gogits/gogs).  It is currently functional for a development install (e.g., minikube).

### CODE REVIEW

**Comment by spxtr** [comment]
@k8s-bot ok to test

**Comment by tvon** [comment]
Is this failing due to [this helm 2.1 bug](https://github.com/kubernetes/helm/issues/1807)?  Can/should the e2e test be updated for helm 2.2?

File: incubator/gogs/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## Configuration
+
+The following tables lists the configurable parameters of the Gogs chart and
+their default values.
+
+| Parameter                        | Description                                                  | Default                                                    |
+| -----------------------          | ----------------------------------                           | ---------------------------------------------------------- |
+| `image.repository`               | Gogs image                                                   | `gogs/gogs`                                                |
+| `image.tag`                      | Gogs image version                                           | `0.9.141`                                                  |
```
Comment (unknwon) [inline incubator/gogs/README.md]: `0.10 RC` just release, probably should use that?

File: incubator/gogs/values.yaml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  sshPort: 22
+
+  ## Gogs configuration values
+  ## ref: https://gogs.io/docs/advanced/configuration_cheat_sheet
+  ##
+  gogs:
+
+    ## Application name, can be your company or team name.
+    ##
+    appName: 'Gogs: Go Git Service'
```
Comment (unknwon) [inline incubator/gogs/values.yaml]: Should probably simplify to just **Gogs**, official repo has removed verbose part, but not a big deal..

File: incubator/gogs/README.md (hunk 1 above)
Comment (tvon) [reply incubator/gogs/README.md]: I'm inclined to leave it as the 0.9.141 as opposed to specifying an RC, and it's an easy thing to override for an installation.

File: incubator/gogs/values.yaml (hunk 2 above)
Comment (tvon) [reply incubator/gogs/values.yaml]: Updated, thanks.

File: incubator/gogs/requirements.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+dependencies:
+  - name: postgresql
+    version: 0.3.0
```
Comment (seanknox) [inline incubator/gogs/requirements.yaml]: Consider bumping this to `0.6.0` (current version of the PostgreSQL chart)

File: incubator/gogs/templates/pvc.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+  annotations:
+  {{- if .Values.persistence.storageClass }}
+    volume.beta.kubernetes.io/storage-class: {{ .Values.persistence.storageClass | quote }}
+  {{- else }}
+    volume.alpha.kubernetes.io/storage-class: default
```
Comment (seanknox) [inline incubator/gogs/templates/pvc.yaml:15]: Thoughts on using `volume.beta.kubernetes.io/storage-class` instead?

**Comment by seanknox** [comment]
Hi @tvon, this looks pretty good. In addition to my comments, this needs `NOTES.txt`. Here's a [good example](https://github.com/kubernetes/charts/blob/master/stable/drupal/templates/NOTES.txt) from the Drupal chart.

File: incubator/gogs/templates/pvc.yaml (hunk 4 above)
Comment (tvon) [reply incubator/gogs/templates/pvc.yaml:15]: Thanks for looking at this. 

I was under the impression that the fallback to `volume.alpha.kubernetes.io/storage-class: default` was the right thing to do if no `storageClass` was provided.  Is this not the case?

**Comment by tvon** [comment]
Note that the failure is the same as it was before, which appears to be a helm 2.1 bug.  Let me know how to proceed on the volume and this should be good to go.

File: incubator/gogs/templates/pvc.yaml (hunk 4 above)
Comment (ekozan) [reply incubator/gogs/templates/pvc.yaml:15]: yeap dont touch this one ;)

**Comment by tvon** [comment]
@seanknox Does this still need changes?

**Comment by wstrange** [comment]
Would really like to see this chart published soon!

Perhaps a future RFE:  For development, deploy using the embedded sqlite3 database engine. It would eliminate the need for postgres and would slim down the footprint.

File: incubator/gogs/templates/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,88 @@
+---
```
Comment (seanknox) [inline incubator/gogs/templates/configmap.yaml]: Can you remove these delimiters from your templates? While they *should* be valid there's an issue with helm that causes problems deleting resources when you delete a release.

File: incubator/gogs/templates/_helpers.tpl [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 24 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 24 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
```
Comment (seanknox) [inline incubator/gogs/templates/_helpers.tpl]: Please bump truncation up to 63 characters.

**Review by seanknox — CHANGES_REQUESTED** [review]

**Comment by seanknox** [comment]
LGTM! Thanks @tvon.

**Review by seanknox — APPROVED** [review]

**Pull request merged**